Sources/SwifterSwift/UIKit/UIImageExtensions.swift

```diff
@@ -32,6 +32,37 @@ public extension UIImage {
         return withRenderingMode(.alwaysTemplate)
     }
 
+    /// SwifterSwift: Average color for this image
+    #if canImport(CoreImage)
+    func averageColor() -> UIColor? {
+        // https://stackoverflow.com/questions/26330924
+        guard let ciImage = CIImage(image: self) else {
+            return nil
+        }
```

Comment:
`UIImage` already has this value as a property. You can also one-line this.
```suggestion
        guard let ciImage = ciImage else { return nil }
```

Reply:
UIImage.ciImage is often nil according to my testing (and googling). Gotta create a new CIImage, unfortunately.

Reply:
Actually yeah, the documentation even says:
> If the UIImage object was initialized using a CGImage, the value of the property is nil.

How about this then?
```suggestion
        guard let ciImage = ciImage ?? CIImage(image: self) else { return nil }
```

Reply:
I think it'll be good to use my suggestion as it can avoid recreating `CIImage` objects if one already exists.

Reply:
👍

Reply:
@gurgeous you marked this as resolved but didn't make the change?